src/uu/cp/src/cp.rs

```diff
@@ -1934,6 +1940,17 @@ fn copy_file(
                 dest.display()
             )));
         }
+        // Fail if cp tries to copy two sources of the same name into a single symlink
+        // Example: "cp file1 dir1/file1 tmp" where "tmp" is a directory containing a symlink "file1" pointing to a file named "foo".
+        // foo will contain the contents of "file1" and "dir1/file1" will not be copied over to "tmp/file1"
+        if copied_destinations.contains(dest) {
+            return Err(Error::Error(format!(
+                "will not copy '{}' through just-created symlink '{}'",
```

Comment:
Apparently there needs to be some other condition, too, because GNU does not always talk about a "just-created symlink", even though it was just created:

```console
$ mkdir src1 src2; touch src1/file src2/file  # setup
$ cp -vs src1/file src2/file .
'src1/file' -> './file'
cp: will not overwrite just-created './file' with 'src2/file'
[$? = 1]
$ rm file
$ cargo run -q cp -vs src1/file src2/file .
'src1/file' -> './file'
cp: will not copy 'src2/file' through just-created symlink './file'
[$? = 1]
```

However, there is no actual difference in behavior, so unless the GNU tests specifically check for this error message, I'd say this error message is good.

Reply:
Interesting, i wonder what other conditions they care about . It does seem like we are really lacking on the edge cases regarding symlinks and copying to directories in general.